lib/ide/src/todo/executor.rs

```diff
@@ -0,0 +1,108 @@
+//! TODO [mwu] This module is still provisional work in progress. Currently
+//!            provided only for tests purposes, needs to be completed and
+//!            properly reviewed.
+//!
+//! Module providing the executor-related types and functions.
+
+#![allow(missing_docs)] // TODO [mwu] remove once module is done
+
+use crate::prelude::*;
+
+use futures::task::LocalSpawnExt;
+use futures::task::LocalSpawn;
+use futures::task::LocalFutureObj;
+use futures::task::SpawnError;
+use futures::executor::LocalPool;
+use futures::executor::LocalSpawner;
+
+use basegl::control::callback::CallbackHandle;
+use basegl::control::EventLoop;
+
+// TODO [mwu] If anything, likely thread local variable should be preferred.
+static mut CURRENT_SPAWNER: Option<Box<dyn LocalSpawn>> = None;
+
+// TODO [mwu] consider whether it is the "current" spawner or rather
+//            "the global" spawner. Is it available from outside the async
+//             context? Do we allow using more than one executor in the IDE?
+#[allow(unsafe_code)]
+pub fn set_global_spawner(spawner: impl LocalSpawn + 'static) {
+    unsafe {
+        CURRENT_SPAWNER = Some(Box::new(spawner));
+    }
+}
+
+#[allow(unsafe_code)]
+pub fn unset_global_spawner() {
+    unsafe {
+        CURRENT_SPAWNER = None;
+    }
+}
+
+#[allow(unsafe_code)]
+pub fn current_spawner() -> &'static mut dyn LocalSpawn {
+    unsafe {
+        CURRENT_SPAWNER.as_mut().expect("no global executor has been provided")
+    }
+}
+
+/// Spawn a task scheduled within a current executor.
+/// Panics, if called when there is no active asynchronous execution.
+pub fn spawn_task(f:impl Future<Output=()> + 'static) {
+    current_spawner().spawn_local(f).ok();
+}
+
+
+
+#[derive(Debug)]
+pub struct JsExecutor {
+    #[allow(dead_code)]
+    executor   : Rc<RefCell<LocalPool>>,
+    #[allow(dead_code)]
+    event_loop : EventLoop,
+    spawner    : LocalSpawner,
+    #[allow(dead_code)]
```

Comment:
what does it do here? I havent seen such usage in the past

Reply:
to get rid of warnings - the purpose of the variable is to keep the value alive, not to "use" it

Reply:
The right way of doing it is to prefix with _. I will fix that

Reply:
Then the name with underscore - you will not get warnings and it indicates exactly this purpose :)